Add links to the CLI reference from step command refs in the main docs

tashian opened the pull request:
Been meaning to add this for a while.

dopey reviewed (approved):
lgtm

tashian committed:
Add links to the CLI reference from step command refs in the main docs

dopey commented on src/pages/docs/step-ca/certificate-authority-server-production.mdx:
```diff
@@ -895,7 +896,7 @@ Be sure the user (in this example, `step`) has write access to the certificate a
 #### Revoking a Certificate
 
-`step ca renew` allows a certificate owner to extend the lifetime of a certificate before it expires.
+[`step ca renew`](/docs/step-cli/reference/ca/init) allows a certificate owner to extend the lifetime of a certificate before it expires.
```
renew

dopey reviewed (approved):
lgtm, I left one comment where the link was going to the wrong command.

tashian committed:
Link fixes

tashian committed:
More link fixes

I used a regex to search & replace:

Search for: (?<!\[)`step (\w+)\s?(\w+)?`
Replace with: [$0](/docs/step-cli/reference/$1/$2)

tashian committed:
Merge branch 'step-links' of https://github.com/smallstep/docs into step-links

tashian committed:
Typo fix

tashian commented:
For posterity, I used a regex to search & replace:

```
Search for: (?<!\[)`step (\w+)\s?(\w+)?`
Replace with: [$0](/docs/step-cli/reference/$1/$2)
```
